crown/src/nockapp/mod.rs

```diff
@@ -138,6 +159,30 @@ impl NockApp {
                 }
                 self.save(permit).await
             },
+            exit = self.exit_recv.recv() => {
+                if let Some(code) = exit {
+                    self.action_channel.close();
+                    let exit_event_num = self.kernel.serf.event_num;
+                    info!("Exit request received, waiting for save for event_num {}", exit_event_num);
+
+                    let mut recv = self.watch_recv.clone();
+                    tokio::task::spawn(async move {
+                        loop {
+                            let _ = recv.changed().await;
+                            let new = *(recv.borrow());
+                            if new == exit_event_num {
+                                info!("Save event_num reached, exiting with code {}", code);
+                                std::process::exit(code as i32);
+                            }
```

Comment:
It would be good for antifragility to `assert!`  that new is not greater than exit_event_num